src/app/services/distance.service.ts

```diff
@@ -0,0 +1,33 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DistanceService {
+  //public dict? : Map<string, { latitude: number; longitude: number }>;
+  public dict = {};
+  constructor(private http: HttpClient) {
+    this.makeDict();
+  }
+  public makeDict() {
+    this.getInfo().subscribe((data) => {
+      const list = data.split('\n');
+      list.shift();
+      list.forEach((e) => {
+        const pair = e.split(`,"`);
```

Comment:
This is somewhat confusing and is apparently done to make parsing the CSV work as it contains a quoted JSON object.

It's probably worthwhile to convert the entire thing to JSON via a one-off script (or potentially just running a regexp in a text editor on the file) rather than do the CSV parsing in the client. I can do this if you want.